rclcpp/src/rclcpp/executor.cpp

```diff
@@ -281,22 +281,42 @@ void
 Executor::execute_subscription(
   rclcpp::SubscriptionBase::SharedPtr subscription)
 {
-  std::shared_ptr<void> message = subscription->create_message();
   rmw_message_info_t message_info;
-
-  auto ret = rcl_take(subscription->get_subscription_handle().get(),
+  // TODO(karsten1987): Can this be set to false directly after initialization?
+  message_info.from_intra_process = false;
+
+  if (subscription->is_raw()) {
+    // TODO(karsten1987): Raise if opensplice !?
```

Comment:
Right now, this function contains a lot of code duplication, given that big `if - else`.

Ideally, we also call `take_raw` to get the still serialized data from the wire and only convert it to a ros message if it's a raw callback. 
The reason I didn't do it this way is that Opensplice doesn't support to get the serialized stream yet and thus would stay incompatible.

Reply:
I think until we make `take_raw` required it has to be this way.